### DISCUSSION TITLE
what is the correct way to get audio with ffmpeg for webrtc.

### DISCUSSION THREAD

**Discussion by serkansmg** [OvenMediaLabs/OvenMediaEngine#1732] [Help]
hi, 
i am very new to oven media engine. 
i am trying to steam webrtc and using ffmpeg inside server. 
i want to stream both llhls and webrtc at the same time.

`ffmpeg -f v4l2 -framerate 15 -video_size 1280x720 -i /dev/video0 \
       -f alsa -i hw:1,0 -c:v libx264 -preset veryfast -b:v 2500k \
       -c:a aac -b:a 128k -ar 44100 -f flv rtmp://localhost:1935/app/live`
this is how i used to stream, i can get video on webrtc but not audio. 
also i can get video but again not video on llhls ..

if i use obs studio. i can get audio from llhls but not from webrtc..
ps: hw:1,0 is correct audio device. i can record audio , ex:
arecord -D hw:1,0 -f cd -t wav -d 5 test_audio.wav

[Server.xml.zip](https://github.com/user-attachments/files/17575685/Server.xml.zip)
this is my server.xml 

and using docker by the way (also installed intel drivers etc.):
version: '3.8'
services:
  ome:
    image: airensoft/ovenmediaengine:0.17.1
    container_name: ome
    environment:
      - OME_HOST_IP=192.168.0.3
    ports:
      - "1935:1935"           # RTMP port
      - "9000:9000"           # Web Dashboard
      - "9999:9999/udp"       # UDP for WebRTC
      - "3333:3333"           # WebRTC signaling
      - "3334:3334"           # WebRTC signaling wss
      - "8080:8080"           #api server
      - "3478:3478"           # STUN/TURN
      - "10000-10009:10000-10009/udp"  # WebRTC media ports
      - "180:80"               # LLHLS HTTP
      - "1443:443"             # LLHLS HTTPS
    volumes:
      - /test/ovenmedia/origin_conf:/opt/ovenmediaengine/bin/origin_conf
      - /test/ovenmedia/edge_conf:/opt/ovenmediaengine/bin/edge_conf
      - /test/ovenmedia/dvr:/opt/ovenmediaengine/dvr
      - /test/ovenmedia/cert:/opt/ovenmediaengine/cert
    devices:
      - /dev/dri:/dev/dri
    restart: unless-stopped
    
    
    so can anyone show me the correct way to get audio also using ffmpeg?

**Comment by dimiden** [comment]
I think it is an issue related to OPUS encoding.
Please refer to https://airensoft.gitbook.io/ovenmediaengine/troubleshooting#id-1.-when-opus-is-not-set-in-the-encoding-options
